packages/angular_devkit/build_angular/src/angular-cli-files/plugins/karma.ts

```diff
@@ -88,12 +88,30 @@ const init: any = (config: any, emitter: any, customFileHandlers: any) => {
 
   config.reporters.unshift('@angular-devkit/build-angular--event-reporter');
 
-  // When using code-coverage, auto-add coverage-istanbul.
-  config.reporters = config.reporters || [];
-  if (options.codeCoverage && config.reporters.indexOf('coverage-istanbul') === -1) {
-    config.reporters.unshift('coverage-istanbul');
+  // When using code-coverage, auto-add karma-coverage.
+  if (options.codeCoverage) {
+    config.plugins = config.plugins || [];
+    config.reporters = config.reporters || [];
+    const {plugins, reporters} = config;
+    const hasCoveragePlugin = plugins.some((p: {}) => 'reporter:coverage' in p);
+    const hasIstanbulPlugin = plugins.some((p: {}) => 'reporter:coverage-istanbul' in p);
+    const hasCoverageReporter = reporters.includes('coverage');
+    const hasIstanbulReporter = reporters.includes('coverage-istanbul');
+    if (hasCoveragePlugin && !hasCoverageReporter) {
+      reporters.push('coverage');
+    }
+    else if (hasIstanbulPlugin && !hasIstanbulReporter) {
+      // coverage-istanbul is deprecated in favor of karma-coverage
+      reporters.push('coverage-istanbul');
+    }
+    else {
+      throw new Error('karma-coverage must be installed in order to run code coverage');
+    }
+    if (hasIstanbulPlugin) {
+      logger.warn(`'karma-coverage-istanbul-reporter' has been deprecated since version 11. Please install 'karma-coverage' instead.`);
```

Comment:
This is misleading Installing karma-coverage will not be enough, users will need to amend their karma config. Maybe we can mention this as well?

Reply:
I've updated the warning to mention updating `karma.conf.js` and added a link to https://github.com/karma-runner/karma-coverage/blob/master/README.md